themes/default/content/blog/kubernetes-sdks-pulumiverse/index.md

```diff
@@ -0,0 +1,194 @@
+
+```yaml
+jobs:
+  find_new_tags:
+    runs-on: ubuntu-latest
+    steps:
+      - ... random boring steps
+
+      - id: find_new_tags
+        run: echo "::set-output name=sdk_versions::$(poetry run python ./bin/find-new-tags.py ${{ github.event.inputs.since_hours }})"
+
+    outputs:
+      sdk_versions: ${{ steps.find_new_tags.outputs.sdk_versions }}
+
+build_nodejs_sdk:
+    needs: find_new_tags
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version: ${{ fromJson(needs.find_new_tags.outputs.sdk_versions) }}
+```
+
+Once our Python script has run and the output stored, we're using a helper function from GitHub called `fromJson` to use the output as a dynamic input to their matrix build system. We'll get exactly one job for every new tag that we've found.
+
+The job to generate the SDK and publish it to npm looks like this:
+
+```yaml
+- name: crd2pulumi-generate-sdk
+  run: |
+    IFS='|' read -r -a build_parts <<< "${{ matrix.version }}"
+    mkdir -p ./_output
+    curl -fsSL -o crd.yaml https://doc.crds.dev/raw/${build_parts[1]}@${build_parts[2]}
+    crd2pulumi --nodejsName ${build_parts[0]} --nodejsPath ./_output ./crd.yaml --force
+    # Fix Package Name
+    sed -ie "s#@pulumi/${build_parts[0]}#@pulumiverse/${build_parts[0]}#g" ./_output/package.json
+    # Fix Package Version
+    sed -ie "s#\"version\": \"\"#\"version\": \"${build_parts[2]}\"#g" ./_output/package.json
+- run: npm publish --access=public
+  working-directory: ./_output
+  env:
+    NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+```
+
+Like I said earlier, the hardwork was already done; we just needed to glue some automation together to improve the DX.
+
+### Publishing
+
+Now that we have this system in place to regularly check and build new SDKs, we needed a home for them to live. Recently, some amazing Pulumi community members, worked together on a new GitHub initiative, with Pulumi support, called Pulumiverse. Pulumiverse aims to be a community lead and governed initiative to provide new abstractions and libraries for consumers of Pulumi to use and make their lives easier. This is the perfect place for these SDKs to live.
+
+You can checkout the [repository today](https://github.com/pulumiverse/kubernetes-sdks).
+
+### SDKs
+
+Currently, the automation publishes NodeJS SDKs (JavaScript and TypeScript) to npm, but plans to add Python will be executed very soon. We're also looking to publish packages for dotNet, but research is still underway.
+
+Sadly, I don't expect to publish packages for Go ... sadly, because Go modules use GitHub for fetching source code - we'd need to publish generated code to the repository and I'm not entirely keen on that approach. I could possibly be persuaded, but I'd love some suggestions for alternate ways to resolve this.
```

Comment:
You use "sadly" twice here, up to you whether or not you wanna find another word